File: downloadFiles/backend/node_service/extractIdVideo.ts

// aqui ele so extrai a id do video. (ex: UrLH86BXQ0E)
export function extractVideoID(url: string): string {
const regex =
  /^(?:https?:\/\/)?(?:www\.)?(?:youtube\.com\/(?:watch\?v=|shorts\/)|youtu\.be\/)([\w-]{11})/;


    const match = url.match(regex);
    if (!match) {
        throw new Error("URL do YouTube é inválida");
    }

    return match[1];
}

File: downloadFiles/backend/pythonservice/api/extractIdVideo.py
import re
def extractVideoId(url: str):
    regex = re.compile(r"^(?:https?:\/\/)?(?:www\.)?(?:youtube\.com\/(?:watch\?v=|shorts\/)|youtu\.be\/)([\w-]{11})")

    match = regex.search(url)
    if not match:
        raise ValueError("URL do YouTube e invalida")

    return match[1];